### ISSUE TITLE
Separate out an admin UI from fedimintd

### ISSUE THREAD

**Issue by jkitman** [fedimint/fedimint#1893]
Currently our setup UI webserver runs as part of `fedimintd` so we need to:
- Add `fedimint` rpc endpoints (with auth) for running setup (DKG)
- Split the webserver into a separate crate that will only need to depend on the API
- As part of a follow-up issue we should add proper tests and decide on a UI framework

Long-term this gives us advantages:
- Could eventually support other front-ends (mobile or federation-specific)
- Admin UI could run locally for added security (doesn't need to be publicly exposed)

See #2003 for discussions on how to build

**Comment by isaack-mungui** [comment]
I'm interested in contributing to Fedimint as part of Summer of Bitcoin. Is this issue and associated issues related to the Admin/Setup UI available for contribution as part of SoB?

**Comment by jkitman** [comment]
@isaack-njama Yes I believe it could be, please ask in our Discord chat https://discord.gg/se87UJPW

**Comment by elsirion** [comment]
> I'm interested in contributing to Fedimint as part of Summer of Bitcoin. Is this issue and associated issues related to the Admin/Setup UI available for contribution as part of SoB?

Would be a nice SoB project indeed! Please propose it, I'll let Adi know, but I think custom proposals were always welcome.

**Comment by isaack-njama** [comment]
@jkitman @elsirion Thank you for the response. I'll make the proposal in the discord server.

**Comment by IgboPharaoh** [comment]
hello @elsirion , I'll love to contribute to this issue

**Comment by okjodom** [comment]
completed after #2559
